bukuserver/forms.py

```diff
@@ -1,18 +1,52 @@
 """Forms module."""
 # pylint: disable=too-few-public-methods, missing-docstring
-from typing import Any, Dict, Tuple
+import re
 from flask_wtf import FlaskForm
-from wtforms.fields import BooleanField, FieldList, StringField, TextAreaField, HiddenField
-from wtforms.validators import DataRequired, InputRequired, ValidationError
-from buku import DELIM, parse_tags
+from wtforms import Form
+from wtforms.fields import BooleanField, FieldList, StringField, TextAreaField, HiddenField, SelectMultipleField
+from wtforms.validators import DataRequired, InputRequired, Length, Regexp, StopValidation
+from buku import DELIM, taglist_str
 from bukuserver import _, _l, LazyString
-from bukuserver.response import Response
 
-def validate_tag(form, field):
+_parse_bool = lambda x: str(x).lower() == 'true'
+
+TAG_RE = re.compile(r'^[^,]*[^,\s]+[^,]*$')
+
+def optional_none(form, field):
+    if field.data is None:
+        raise StopValidation()
```

Comment:
Unlike `DataRequired`/`InputRequired`, this one treats `None` as a valid value (but still skips remaining validators)